scripts/export_tracks.py

```diff
@@ -0,0 +1,116 @@
+import logging.handlers
+import os
+import traceback
+import sys
+from arcgis.gis import GIS
+
+
+def initialize_logging(log_file=None):
+    """
+    Setup logging
+    :param log_file: (string) The file to log to
+    :return: (Logger) a logging instance
+    """
+    # initialize logging
+    formatter = logging.Formatter(
+        "[%(asctime)s] [%(filename)30s:%(lineno)4s - %(funcName)30s()][%(threadName)5s] [%(name)10.10s] [%(levelname)8s] %(message)s")
+    # Grab the root logger
+    logger = logging.getLogger()
+    # Set the root logger logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
+    logger.setLevel(logging.DEBUG)
+    # Create a handler to print to the console
+    sh = logging.StreamHandler(sys.stdout)
+    sh.setFormatter(formatter)
+    sh.setLevel(logging.INFO)
+    # Create a handler to log to the specified file
+    if log_file:
+        rh = logging.handlers.RotatingFileHandler(log_file, mode='a', maxBytes=10485760)
+        rh.setFormatter(formatter)
+        rh.setLevel(logging.DEBUG)
+        logger.addHandler(rh)
+    # Add the handlers to the root logger
+    logger.addHandler(sh)
+    return logger
+
+
+def main(arguments):
+    logger = initialize_logging(arguments.log_file)
+    logger.info("Authenticating...")
+    # Authenticate to ArcGIS Online
+    gis = GIS(arguments.org_url,
+              username=arguments.username,
+              password=arguments.password,
+              verify_cert=not arguments.skip_ssl_verification)
+    if gis.properties['isPortal']:
+        raise Exception("Export is not supported for the location tracking service with ArcGIS Enterprise")
+    logger.info("Exporting...")
+    tracks_item = gis.content.get(args.tracks_item)
+    # Create date range using track age
+    # Always export up through the last full day (intentionally excludes part of current day)
+    start_date = pendulum.today(args.time_zone) - datetime.timedelta(days=args.track_age)
+    end_date = pendulum.today(args.time_zone).at(23, 59, 59) - datetime.timedelta(days=1)
+    # Export the tracks
+    name = f"tracks_{start_date.to_date_string()}_{end_date.to_date_string()}"
+    csv_item = tracks_item.export(name,
+                                  export_format='CSV',
+                                  parameters={
+                                      "layers": [
+                                          {
+                                              "id": 0,
+                                              "where": f"location_timestamp <= '{end_date.in_tz('UTC').to_datetime_string()}' AND "
+                                                       f"location_timestamp >= '{start_date.in_tz('UTC').to_datetime_string()}'"
+                                          }
+                                      ]
+                                  }
+                                  )
+    logger.info("Downloading...")
+    # Download the CSV file
+    save_path = os.path.abspath(args.output_directory)
```

Comment:
add handling if output dir is not found or if download is unsuccessful

Reply:
I added a check for the directory. The API will raise an exception if the download/export fails.